src/main_test.py

```diff
@@ -44,6 +52,24 @@ def testRoutesInheritance(self):
     self._VerifyInheritance(main._CRON_ROUTES, handlers.BaseCronHandler)
     self._VerifyInheritance(main._TASK_ROUTES, handlers.BaseTaskHandler)
 
+  def testStrictHandlerMethodRouting(self):
+    """Checks that handler functions properly limit applicable HTTP methods."""
+    router = webapp2.Router(main._USER_ROUTES + main._AJAX_ROUTES +
+                            main._ADMIN_ROUTES + main._ADMIN_AJAX_ROUTES)
+    routes = router.match_routes + router.build_routes.values()
+    while routes:
+      route = routes.pop()
+      if issubclass(route.__class__, webapp2_extras.routes.MultiRoute):
+        routes += list(route.get_routes())
+        continue
+
+      if issubclass(route.handler, webapp2.RedirectHandler):
+        continue
+
+      if route.handler_method and not route.methods:
```

Comment:
I realized this after submitting this test, but it might be nicer to actually aggregate these all into a list and then fail at the end if the list isn't empty (this way we get all the routes / handlers listed that need closer inspection instead of failing on the first one).  Would you mind adding that behavior?